src/aiidalab_qe/app/submission/global_settings/model.py

```diff
@@ -125,12 +123,9 @@ def check_resources(self):
         num_sites = len(self.input_structure.sites)
         volume = self.input_structure.get_cell_volume()
 
-        try:
-            localhost_cpus = len(os.sched_getaffinity(0))
-        except Exception:
-            # Fallback, in some OS os.sched_getaffinity(0) is not supported
-            # However, not so reliable in containers
-            localhost_cpus = os.cpu_count()
+        machine_cpus = orm.load_node(
+            pw_code_model.selected
+        ).computer.get_default_mpiprocs_per_machine()
```

Comment:
```suggestion
        code = orm.load_node(
            pw_code_model.selected
        )
        machine_cpus = code.computer.get_default_mpiprocs_per_machine()
```

Could be more readable.